src/main/java/org/store/springbootbookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,104 @@
+package org.store.springbootbookstore.service.impl;
+
+import jakarta.transaction.Transactional;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+import org.store.springbootbookstore.dto.order.CreateOrderRequestDto;
+import org.store.springbootbookstore.dto.order.OrderResponseDto;
+import org.store.springbootbookstore.dto.order.PatchOrderRequestDto;
+import org.store.springbootbookstore.dto.orderitem.OrderItemResponseDto;
+import org.store.springbootbookstore.exception.EntityNotFoundException;
+import org.store.springbootbookstore.mapper.OrderItemMapper;
+import org.store.springbootbookstore.mapper.OrderMapper;
+import org.store.springbootbookstore.model.Order;
+import org.store.springbootbookstore.model.OrderItem;
+import org.store.springbootbookstore.model.ShoppingCart;
+import org.store.springbootbookstore.model.User;
+import org.store.springbootbookstore.repository.OrderItemRepository;
+import org.store.springbootbookstore.repository.OrderRepository;
+import org.store.springbootbookstore.repository.ShoppingCartRepository;
+import org.store.springbootbookstore.service.OrderService;
+import org.store.springbootbookstore.service.ShoppingCartService;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+    private final ShoppingCartService shoppingCartService;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final OrderItemRepository orderItemRepository;
+
+    @Override
+    public Page<OrderResponseDto> getAllForCurrentUser(Pageable pageable) {
+        Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+        User user = (User) auth.getPrincipal();
+        return orderRepository.findAllByUserId(user.getId(), pageable)
+                .map(orderMapper::toDto);
+    }
+
+    @Override
+    @Transactional
+    public OrderResponseDto createOrder(CreateOrderRequestDto requestDto) {
+        ShoppingCart shoppingCart = shoppingCartService.getCartForCurrentUser();
+        if (shoppingCart.getCartItems().isEmpty()) {
+            throw new EntityNotFoundException("No shopping cart items found");
```

Comment:
add parameter to exception 
```suggestion
            throw new OrderProcessingException("No shopping cart items found");
```